scripts/cc-cli/src/utils/account.ts

```diff
@@ -14,69 +15,83 @@ export function initECDSAKeyringPairFromPK(pk: string) {
   return pair;
 }
 
-export async function getStashSeedFromEnvOrPrompt(interactive: boolean) {
+export async function initStashKeyring(
+  options: OptionValues,
+): Promise<KeyringPair> {
   try {
-    return await getSeedFromEnvOrPrompt("CC_STASH_SEED", "stash", interactive);
+    return await initKeyringFromEnvOrPrompt("CC_STASH_SEED", "stash", options);
   } catch (e) {
     console.error(getErrorMessage(e));
     process.exit(1);
   }
 }
-export async function getControllerSeedFromEnvOrPrompt(interactive: boolean) {
+
+export async function initControllerKeyring(
+  options: OptionValues,
+): Promise<KeyringPair> {
   try {
-    return await getSeedFromEnvOrPrompt(
+    return await initKeyringFromEnvOrPrompt(
       "CC_CONTROLLER_SEED",
       "controller",
-      interactive,
+      options,
     );
   } catch (e) {
     console.error(getErrorMessage(e));
     process.exit(1);
   }
 }
-export async function getCallerSeedFromEnvOrPrompt(interactive: boolean) {
+
+export async function initCallerKeyring(
+  options: OptionValues,
+): Promise<KeyringPair> {
   try {
-    return await getSeedFromEnvOrPrompt("CC_SEED", "caller", interactive);
+    return await initKeyringFromEnvOrPrompt("CC_SEED", "caller", options);
   } catch (e) {
     console.error(getErrorMessage(e));
     process.exit(1);
   }
 }
 
-async function getSeedFromEnvOrPrompt(
-  envVar = "CC_SEED",
-  accountRole = "caller",
-  interactive = true,
-) {
+export async function initKeyringFromEnvOrPrompt(
+  envVar: string,
+  accountRole: string,
+  options: OptionValues,
+): Promise<KeyringPair> {
+  const interactive = options.input;
+  const ecdsa = options.ecdsa;
+  envVar = ecdsa ? "CC_PK" : envVar;
+  const inputName = ecdsa ? "private key" : "seed phrase";
+  const validateInput = ecdsa ? () => true : mnemonicValidate;
```

Comment:
No input validation when ecdsa = true. 

Q: Are there functions to validate a private key  ? We could at least validate the length .

Reply:
How do I create / generate a key in this ecdsa format ?